packages/dds/tree/src/test/forestTestSuite.ts

```diff
@@ -0,0 +1,724 @@
+
+            const expectedParent: UpPath = {
+                parent: undefined,
+                parentField: rootFieldKeySymbol,
+                parentIndex: 0,
+            };
+
+            assert.deepStrictEqual(parentPath, expectedParent);
+            assert.deepStrictEqual(parentPath2, expectedParent);
+
+            const expectedChild1: UpPath = {
+                parent: expectedParent,
+                parentField: brand("data"),
+                parentIndex: 0,
+            };
+
+            const expectedChild2: UpPath = {
+                parent: expectedParent,
+                parentField: brand("data"),
+                parentIndex: 1,
+            };
+
+            assert.deepStrictEqual(childPath1, expectedChild1);
+            assert.deepStrictEqual(childPath2, expectedChild2);
+
+            assert.equal(forest.tryMoveCursorToNode(parentAnchor, cursor), TreeNavigationResult.Ok);
+            assert.equal(cursor.value, undefined);
+            assert.equal(forest.tryMoveCursorToNode(childAnchor1, cursor), TreeNavigationResult.Ok);
+            assert.equal(cursor.value, 1);
+            assert.equal(forest.tryMoveCursorToNode(childAnchor2, cursor), TreeNavigationResult.Ok);
+            assert.equal(cursor.value, 2);
+
+            // Cleanup is not required for this test (since anchor set will go out of scope here),
+            // But make sure it works:
+            forest.anchors.forget(parentAnchor);
+            forest.anchors.forget(childAnchor1);
+            forest.anchors.forget(childAnchor2);
+            forest.anchors.forget(parentAnchor2);
+            assert(forest.anchors.isEmpty());
+        });
+
+        it("using an anchor that went away returns NotFound", () => {
+            const forest = factory(new InMemoryStoredSchemaRepository(defaultSchemaPolicy));
+            const dependent = new MockDependent("dependent");
+            recordDependency(dependent, forest);
+
+            const content: JsonableTree[] = [
+                {
+                    type: jsonObject.name,
+                    fields: {
+                        data: [
+                            { type: jsonNumber.name, value: 1 },
+                            { type: jsonNumber.name, value: 2 },
+                        ],
+                    },
+                },
+            ];
+            initializeForest(forest, content.map(singleTextCursor));
+
+            const cursor = forest.allocateCursor();
+            moveToDetachedField(forest, cursor);
+            assert(cursor.firstNode());
+            cursor.enterField(brand("data"));
+            cursor.enterNode(0);
+            const firstNodeAnchor = cursor.buildAnchor();
+            cursor.clear();
+
+            const mark: Delta.Delete = { type: Delta.MarkType.Delete, count: 1 };
+            const delta: Delta.Root = new Map([[rootFieldKeySymbol, [0, mark]]]);
+            // TODO: make type-safe
```

Comment:
What's wrong with the type here?

Reply:
Looks like there's a whole bunch of these.

Reply:
Well, I don't see any issues with the types here so I'd remove the TODOs, but I understand if you'd rather not. Doesn't bother me.

Reply:
I'll leave them for now and ask Craig later if they can be removed, just in case he had something specific in mind with this todo.

Reply:
These are obsolete and date from before we had typing for detached fields as field keys sorted out. Should be removed.

Reply:
This is also an example of a case of where code is moved and updated in the same PR making review more difficult. Not sure it would be worth it in this case, but it might have been better to update and move the test suite in separate changes. That said, we caught this because of rereview of code that was not new, so that has its value too!